pyteal/ast/subroutine.py

```diff
@@ -573,6 +573,15 @@ def __call__(
     def name(self) -> str:
         return self.subroutine.name()
 
+    def method_signature(self) -> str:
+        if not self.is_registrable():
+            raise TealInputError(
+                "Only registrable methods may return a method signature"
+            )
+
+        args = [str(v) for v in self.subroutine.abi_args.values()]
```

Comment:
the purpose of `ABIReturnSubroutine` is to allow for returning `ABI` types, not specifying all input types to be `ABI` types. So what you have suggested seems restraining the use of `ABIReturnSubroutine` for arg take in...?

Reply:
Ok... finally I get it. Thanks for your patience!!!

So now I'm going to change my point to a very minor `nit`. I think either `is_registrable(abi_method)` should be moved into the `Router` class and/or be renamed to `is_abi_routable()`. Unless we have another use case of the word `register`, I don't think we should add another synonym. This would cause some confusion (as apparently happened to me).

Reply:
good point!!!! I am terrible at naming stuffs, `is_abi_routable()` in `ABIReturnSubroutine` is a good option to me, will change in 1 sec